src/lib/address/address.ts

```diff
@@ -134,6 +148,15 @@ export type ParseAddressInputReturn = {
 export type ParseAddressInput = (args: ParseAddressInputArgs) => Promise<ParseAddressInputReturn>
 
 export const parseAddressInput: ParseAddressInput = async args => {
+  const isYat = await validateYat(args)
+  if (isYat) {
+    const address = await resolveYat(args)
+    const vanityAddress = await reverseLookupVanityAddress({
+      chainId: args.chainId,
+      value: address,
+    })
+    return { address, vanityAddress }
+  }
```

Comment:
I don't think this is the best place for this:

- this isn't using our resolvers abstractions and we effectively have a branch that's tied to a specific resolver - this should live in `vanityResolversByChainId`
- by not using the abstraction, we are effectively not having a resolver defining its supported chainId(s). While this looks like this works fine for ETH sends, this is what happens when you input a Y.at in native wallet sends for e.g UTXOs:

<img width="483" alt="image" src="https://user-images.githubusercontent.com/17035424/185895013-e92cffd1-8b8d-4c3f-885a-a2fec4183761.png">
<img width="468" alt="image" src="https://user-images.githubusercontent.com/17035424/185895062-1abba362-4188-4a8e-9d51-3b27deb7a6d7.png">
<img width="391" alt="image" src="https://user-images.githubusercontent.com/17035424/185895199-9cac6a4a-311f-48fb-8750-5caa9ec2ca31.png">